snap-packages/from-source/snapcraft-template.yaml

```diff
@@ -0,0 +1,78 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: @SNAP_NAME@
+
+summary: @SNAP_SUMMARY@ 
+description: |@SNAP_DISCLAIMER@
+  Apache NetBeans IDE lets you quickly and easily develop Java
+  desktop, enterprise, and  web applications, as well as HTML5 applications
+  with HTML, JavaScript, and CSS. The IDE also provides a great set of tools for
+  PHP and C/C++ developers.
+  It is free and open source and has a large community of users and developers
+  around the world.
+  
+  It requires Java 8 or later Java Development Kit installed.
+
+icon: snap/gui/frame512.png
+confinement: classic
+grade: @SNAP_GRADE@
+base: core18
+architectures: [ amd64 ]
+compression: lzo
+assumes:
+  - command-chain
+version: "@SNAP_VERSION@"
+
+parts:
+      
+  build:
+    build-attributes: [ no-patchelf ]
+    build-packages:
+      - git
+      - unzip
+      - openjdk-11-jdk-headless
+    plugin: ant
+    source: https://github.com/apache/netbeans.git
+    source-branch: "@SNAP_BRANCH@"
+    filesets:
+        netbeans: [ netbeans/*, -netbeans/*.built, -netbeans/extra ]
+    override-build: |
+        export JAVA_HOME="/usr/lib/jvm/java-11-openjdk-amd64"
+        ant -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false @BUILD_NUMBER@ -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json
+        mv nbbuild/netbeans $SNAPCRAFT_PART_INSTALL/netbeans
```

Comment:
This is using the `nbbuild/netbeans` folder rather than the zip from `ant build`. This is probably why you were picking up the OpenJFX binaries / extra folder?  Should be using the binary zip as some things are filtered out in its creation.

Reply:
Well a new commit has been provided for building snap from the zip distribution, however netbeans/extra is filtered out  by now in the fileset on line 52

Reply:
As long as it replicates the output from `build` and `zip-cluster-config` either is probably fine. https://github.com/apache/netbeans/blob/master/nbbuild/build.xml#L673